### PR TITLE
Update of dependencies in go.mod: zap, bbolt, grpc, ...

### PR DESCRIPTION
**Pull request by ptabor**
- go.uber.org/zap@v1.17.0
      - bbolt-1.3.6
      - grpc -1.38
      - github.com/coreos/go-systemd/v22 v22.3.2
      - github.com/google/btree v1.0.1
      - github.com/json-iterator/go v1.1.11
      - github.com/mattn/go-runewidth v0.0.13
      - github.com/prometheus/client_golang v1.10.0
      - github.com/sirupsen/logrus v1.8.1

### CODE REVIEW

File: server/go.mod [hunk 1]
```diff
@@ -59,7 +59,9 @@ replace (
...
-replace (
-	go.etcd.io/etcd => ./FORBIDDEN_DEPENDENCY
-	go.etcd.io/tests/v3 => ./FORBIDDEN_DEPENDENCY
-)
+replace go.etcd.io/etcd => ./FORBIDDEN_DEPENDENCY
+
+replace go.etcd.io/tests/v3 => ./FORBIDDEN_DEPENDENCY
+
+// Avoid dependency on go.etcd.io/etcd.
+replace github.com/go-kit/kit v0.10.0 => github.com/go-kit/kit v0.10.1-0.20210530035833-3b053a73c0f4
```
Comment (lilic) [inline server/go.mod:67]: Do we still need this? I thought it was solved by todays prometheus/common release? Or does it not come from client_golang?

**Review by lilic — COMMENTED** [review]
Thank you! 🎉

File: server/go.mod (hunk 1 above)
Comment (ptabor) [reply server/go.mod:67]: It comes from `prometheus/client_golang`
I might try to depend on https://github.com/prometheus/client_golang/commit/253906201bda760621fa671fa1541a4ac3df29bd instead.

Comment (lilic) [reply server/go.mod:67]: I see that was not released yet, makes sense. @kakkoyun is there any plans to maybe release a patch release? Thank you!

Comment (ptabor) [reply server/go.mod]: Thank you for the suggestion. As in both scenarios we depend on unstable version, I found depending on ustable `prometheus/client_golang` cleaner. PTAL.

**Review by lilic — APPROVED** [review]
lgtm

File: server/go.mod (hunk 1 above)
Comment (kakkoyun) [reply server/go.mod]: No plans yet for a need patch release. I'd advise to use a master version :blush:

Comment (ptabor) [reply server/go.mod]: It means that for release-3.5 we need to stick to last stable version that does not introduce a cyclic dependency, i.e: 
github.com/prometheus/client_golang v1.5.1 (https://github.com/etcd-io/etcd/blob/fea5ace182a74c29da64c8283ea0a84203c2d3f2/client/v3/go.mod#L8)

Comment (lilic) [reply server/go.mod]: Not sure how the k8s deps would play with that?

**Pull request merged**